### PR TITLE
Make `o` do the same as `ṅ` but starting at `1` instead of `0`

### LINKED ISSUE #511: Way to get first n *positive* integers

### PR DESCRIPTION
**Pull request by Lyxal**
Closes #511

### CODE REVIEW

**Review by PyGamer0 — APPROVED** [review]

**Review by Tams-Tams — APPROVED** [review]

File: documents/knowledge/elements.yaml [hunk 1]
```diff
@@ -1149,6 +1149,8 @@
   arity: 2
   overloads:
     any-any: a.replace(b,"")
+    num-fun: first a positive integers where b is truthy
+    fun-num: first b positive integers where a is truthy
```
Comment (cgccuser) [inline documents/knowledge/elements.yaml:1153]:
```suggestion
    num-fun: first a positive integers where b is truthy
    fun-num: first b positive integers where a is truthy
    any-any: a.replace(b,"")
```
It just seems like the any-any overloads is matched before the num-fun/fun-num overloads right now.

File: vyxal/elements.py [hunk 2]
```diff
@@ -2854,9 +2856,24 @@ def regex_sub(lhs, rhs, other, ctx):
 def remove(lhs, rhs, ctx):
     """Element o
     (any, any) -> a.remove(b)
+    (fun, num) -> first b positive integers where a is truthy
```
Comment (cgccuser) [inline vyxal/elements.py:2859]: This doesn't have the (num, fun) overload. Also, could the (any, any) overload be moved to the bottom here too?

File: vyxal/elements.py [hunk 3]
```diff
@@ -2854,9 +2856,24 @@ def regex_sub(lhs, rhs, other, ctx):
...
     """
     lhs = iterable(lhs)
     ts = vy_type(lhs)
+    if set(vy_type(lhs, rhs)) == {types.FunctionType, NUMBER_TYPE}:
+        lhs, rhs = (rhs, lhs) if ts is types.FunctionType else (lhs, rhs)
```
Comment (cgccuser) [inline vyxal/elements.py:2864]: We really need a better way to check types and swap than this :\. Not a big deal though, you can resolve this comment and we can work on it later

**Review by cgccuser — CHANGES_REQUESTED** [review]
The code itself is good, just a few minor suggestions

File: documents/knowledge/elements.yaml (hunk 1 above)
Comment (Tams-Tams) [reply documents/knowledge/elements.yaml:1153]: What do you think of a file containing the order of overloads?

Comment (cgccuser) [reply documents/knowledge/elements.yaml:1153]: `elements.yaml` has that already

File: vyxal/elements.py (hunk 2 above)
Comment (cgccuser) [inline vyxal/elements.py:2859]:
```suggestion
    (num, fun) -> first a positive integers where b is truthy
    (fun, num) -> first b positive integers where a is truthy
    (any, any) -> a.remove(b)
```

**Review by cgccuser — CHANGES_REQUESTED** [review]

**Review by cgccuser — APPROVED** [review]

**Pull request merged**